features/update/basic/feature.ts

```diff
@@ -0,0 +1,53 @@
+import { WorkflowUpdateFailedError } from '@temporalio/client';
+import { Feature } from '@temporalio/harness';
+import * as wf from '@temporalio/workflow';
+import * as assert from 'assert';
+
+const WORKFLOW_INITIAL_STATE = '';
+const UPDATE_ARG = 'update-arg';
+const BAD_UPDATE_ARG = 'reject-me';
+const UPDATE_RESULT = 'update-result';
+const UPDATE_RESULT_NOT_YET_RECEIVED = 'update-result-not-yet-received';
+
+const myUpdate = wf.defineUpdate<string, [string]>('myUpdate');
+
+/**
+ * A workflow with a signal and signal validator. If accepted, the signal makes
+ * a change to workflow state. The workflow does not terminate until such a
+ * change occurs.
+ */
+export async function workflow(): Promise<string> {
+  let state = WORKFLOW_INITIAL_STATE;
+  const update = (arg: string) => {
+    state = arg;
+    return UPDATE_RESULT;
+  };
+  const validate = (arg: string) => {
+    if (arg == BAD_UPDATE_ARG) {
+      throw new Error('Invalid Update argument');
+    }
+  };
+  wf.setHandler(myUpdate, update, validate);
```

Comment:
Can you confirm the signature of `setHandler` here? I just want to confirm that the update can return the value or a promise of the value.

Reply:
Confirmed: the handler can be an async function or a normal function. But this was a good call -- when you made this comment the PR did not demonstrate both.